### DISCUSSION TITLE
Error: Throttling: Too many requests, HTTP status: TooManyRequests

### DISCUSSION THREAD

**Discussion by geirolsen** [microsoft/semantic-kernel#465] [Q&A]
I created the first example that includes: "1st Law of Thermodynamics - Energy cannot be created or destroyed." 
I used my OpenAI key. 
I setup an OpenAI account that has monies associated with it. 
The title of this discussion states the error message I'm seeing. Assuming I missed something very trivial. ChatGPT doesn't know. 
Not obvious what I missed so I'm asking the question here. 

Thanks
--Geir

**Comment by daniellogangill** [comment]
I've received the same error in the Design Chain when trying to execute Step 3 - 'Turn the empathy insights into problem definitions'.  Any guidance on what I'm doing wrong or what's causing the error would be greatly appreciated.

Thanks
Daniel

**Comment by dluc** [comment]
hi @geirolsen @daniellogangill you're not doing anything wrong. OpenAI popularity means that sometimes they have too many requests to deal with and they return that error, asking users to retry. I would suggest designing the code to retry, e.g. wait 1-2secs and retry, or consider using Azure OpenAI which comes with an SLA to guarantee some bandwidth is always available.

**Reply by daniellogangill** [reply]
Thanks @dluc - Really appreciated and very helpful.  Shall give it a go.  

Thanks
Daniel

**Reply by mtnsh** [reply]
hi, i was waiting for a few seconds and replaced to chat gpt, yet get same error - 

> Error: Throttling: Too many requests, HTTP status: 429

//Alternative using OpenAI
kernel.Config.AddOpenAITextCompletionService("chat",
"gpt-3.5-turbo", // OpenAI Model name

 this is all i was changing in the initial program (from azure to openai) - am i missing something?
thanks

p.s - i didnt use the azure route yet, however, i do pay the monthly subscription to openai if that matters
